lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -491,17 +499,45 @@ instance NFData Address
 instance Buildable Address where
     build = build . toText
 
+instance ToText Address where
+    toText = T.decodeUtf8
+        . convertToBase Base16
+        . getAddress
+
 instance FromText Address where
-    fromText x = maybe
-        (Left $ TextDecodingError err)
-        (pure . Address)
-        (decodeBase58 bitcoinAlphabet $ T.encodeUtf8 x)
+    fromText = bimap textDecodingError Address
+        . convertFromBase Base16
+        . T.encodeUtf8
```

Comment:
I've kept the `FromText` and `ToText` as they are conveniently used to serialize addresses to disk, but went for an agnostic `Base16` encoding here. This isn't user-facing, so there's no reason to pick an heavier encoding format like bech32 or base58 for this.

Reply:
I would even go and put this comment in the code under `-- NOTE: ...` as other devs might be interested in the encoding choice